File: Runners/JavaCodeRunner/src/main/java/csc/coderunner/CodeTestResult.java
package csc.coderunner;

/**
 * The result of running a test.
 */
public class CodeTestResult
{
	private String name;
	private boolean completed;
	private String exception;
	private String returnValue;
	private String output;
	
	/**
	 * Constructor.
	 * @param name The name of the test.
	 * @param completed Whether or not the test completed successfully.
	 * @param exception The exception thrown by the test, if it did not complete successfully.
	 * @param returnValue The return value of the test, if the test method had a non-void return type.
	 * @param output The output of the test, if any.
	 */
	public CodeTestResult(
		String name, 
		boolean completed, 
		String exception, 
		String returnValue, 
		String output)
	{
		this.name = name;
		this.completed = completed;
		this.exception = exception;
		this.returnValue = returnValue;
		this.output = output;
	}

	/**
	 * @return The name of the test.
	 */
	public String getName()
	{
		return name;
	}

	/**
	 * @return Whether or not the test completed successfully.
	 */
	public boolean getCompleted()
	{
		return completed;
	}
	
	/**
	 * @return The exception thrown by the test, if it did not complete successfully.
	 */
	public String getException()
	{
		return exception;
	}
	
	/**
	 * @return The return value of the test, if the test method had a return type.
	 */
	public String getReturnValue()
	{
		return returnValue;
	}
	
	/**
	 * @return The output of the test, if any.
	 */
	public String getOutput()
	{
		return output;
	}
}

File: Services/src/BuildService/BuildService.Model/CodeRunner/CodeTestResult.cs
namespace CSC.BuildService.Model.CodeRunner
{
	/// <summary>
	/// The result of a test.
	/// </summary>
	public class CodeTestResult
	{
		/// <summary>
		/// The name of the test.
		/// </summary>
		public string Name { get; set; }

		/// <summary>
		/// Whether or not the test completed successfully.
		/// </summary>
		public bool Completed { get; set; }

		/// <summary>
		/// The exception, if the test did not complete successfully.
		/// </summary>
		public string Exception { get; set; }

		/// <summary>
		/// The return value of the test, if the test method had a non-void return type.
		/// </summary>
		public string ReturnValue { get; set; }

		/// <summary>
		/// The output of the test, if any.
		/// </summary>
		public string Output { get; set; }
	}
}